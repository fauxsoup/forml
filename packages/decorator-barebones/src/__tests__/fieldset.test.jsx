import FieldSet from '../fieldset';
import Context from '@forml/context';
import React from 'react';
import { render } from '@testing-library/react';
import * as decorator from '../';

describe('renders', function () {
    let form;
    let title = 'title';
    let description = 'description';

    beforeEach(function () {
        form = { type: 'fieldset', items: [{ key: [] }] };
    });

    test('with title and description', function () {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <FieldSet form={form} title={title} description={description} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test('with title and no description', function () {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <FieldSet form={form} title={title} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test('with description and no title', function () {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <FieldSet form={form} description={title} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test('with no title or description', function () {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <FieldSet form={form} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });
});
