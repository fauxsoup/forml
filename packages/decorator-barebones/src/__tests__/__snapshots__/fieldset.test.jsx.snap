// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with description and no title 1`] = `
<div>
  <fieldset>
    <p>
      title
    </p>
    <div />
  </fieldset>
</div>
`;

exports[`renders with no title or description 1`] = `
<div>
  <fieldset>
    <div />
  </fieldset>
</div>
`;

exports[`renders with title and description 1`] = `
<div>
  <fieldset>
    <legend>
      title
    </legend>
    <p>
      description
    </p>
    <div />
  </fieldset>
</div>
`;

exports[`renders with title and no description 1`] = `
<div>
  <fieldset>
    <legend>
      title
    </legend>
    <div />
  </fieldset>
</div>
`;
